cli/commands/project_sync.go

```diff
@@ -0,0 +1,235 @@
+  err := exec.Command("docker","container", "run", "--detach", "--rm",
+    "-v", fmt.Sprintf("%s:/unison", volumeName),
+    "-e", "UNISON_DIR=/unison",
+    "-l", fmt.Sprintf("com.dnsdock.name=%s", volumeName),
+    "-l", "com.dnsdock.image=volume.outrigger",
+    "--name", volumeName,
+    "outrigger/unison:latest",
+  ).Run()
+  if err != nil {
+  	cmd.out.Error.Fatalf("Error starting sync container %s: %v", volumeName, err)
+  }
+
+  var ip = cmd.WaitForUnisonContainer(volumeName)
+
+  cmd.out.Info.Println("Initializing sync")
+
+  // Remove the log file, the existence of the log file will mean that sync is up and running
+  var logFile = fmt.Sprintf("%s.log", volumeName)
+  exec.Command("rm", "-f", logFile).Run()
+
+  unisonArgs := []string{
+    ".",
+    fmt.Sprintf("socket://%s:%d/", ip, UNISON_PORT),
+    "-auto", "-batch", "-silent", "-contactquietly",
+    "-repeat", "watch",
+    "-prefer", ".",
+    "-logfile", logFile,
+    "-ignore", "Name .git",
+    "-ignore", fmt.Sprintf("Name %s", logFile),
+  }
+  // Append ProjectConfig ignores here
+  for _, ignore := range config.Sync.Ignore {
+    unisonArgs = append(unisonArgs, "-ignore", ignore)
+  }
+
+  cmd.out.Verbose.Printf("Unison Args: %s", strings.Join(unisonArgs[:]," "))
+  if err = exec.Command("unison", unisonArgs...).Start(); err != nil {
+    cmd.out.Error.Fatalf("Error starting local unison process: %v", err)
+  }
+
+  cmd.WaitForSyncInit(logFile)
+
+  return nil
+}
+
+// Start the unison sync process
+func (cmd *ProjectSync) RunStop(ctx *cli.Context) error {
+  project.ConfigInit()
+  config := project.GetProjectConfigFromFile(project.GetConfigPath())
+  volumeName := cmd.GetVolumeName(ctx, config)
+  cmd.out.Verbose.Printf("Stopping sync with volume: %s", volumeName)
+
+  cmd.out.Info.Println("Stopping unison container")
+  exec.Command("docker","container", "stop", volumeName).Run()
+
+  return nil
+}
+
+
+// Find the volume name through a variety of fall backs
+func (cmd *ProjectSync) GetVolumeName(ctx *cli.Context, config project.ProjectConfig) string {
+  // 1. Check for argument
+  if ctx.Args().Present() {
+    return ctx.Args().First()
+  }
+
+  // 2. Check for config
+  if config.Sync != nil && config.Sync.Volume != "" {
+    return config.Sync.Volume
+  }
+
+  // 3. Parse compose file looking for an external volume named *-sync
+  if composeConfig, err := cmd.LoadComposeFile(); err == nil {
+    for name, volume := range composeConfig.Volumes {
+      if strings.HasSuffix(name, "-sync") && volume.External{
+        return name
+      }
+    }
+  }
+
+  // 4. Use local dir for the volume name
+  if dir, err := os.Getwd(); err == nil {
+    var _, folder = path.Split(dir)
+    return folder
+  } else {
+    cmd.out.Error.Println(err)
+  }
+
+  cmd.out.Error.Fatal("Unable to determine a name for the sync volume")
+  return ""
+}
+
+// Load the proper compose file
+func (cmd *ProjectSync) LoadComposeFile() (*ComposeFile, error) {
```

Comment:
I tried. Non of the libraries include either the most recent version of compose (v3) or the "overriding rules" that come from multiple files. They just dont have compose file parsing such that it is a library to use at the moment. Happy to transition over to one when it exists though.